src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -224,7 +219,7 @@ public StudentParticipation startExercise(Exercise exercise, String username) {
             // we might need to perform an empty commit (depends on the CI system), we perform this here, because it should not trigger a new programming submission
             programmingExerciseStudentParticipation = performEmptyCommit(programmingExerciseStudentParticipation);
             // Note: we configure the repository webhook last, so that the potential empty commit does not trigger a new programming submission (see empty-commit-necessary)
-            programmingExerciseStudentParticipation = configureRepositoryWebHook(programmingExerciseStudentParticipation);
+            versionControlService.get().addWebHookForParticipation(programmingExerciseStudentParticipation);
```

Comment:
we should keep this in the below `configureRepositoryWebHook` method because it contains some additional logic that is necessary in case users invoke start exercise multiple times simultaneously